packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -74,13 +120,38 @@ export class EventSourceMapping extends cdk.Resource {
       throw new Error(`maxBatchingWindow cannot be over 300 seconds, got ${props.maxBatchingWindow.toSeconds()}`);
     }
 
+    if (props.maxRecordAge && (props.maxRecordAge.toSeconds() < 60 || props.maxRecordAge.toDays({integral: false}) > 7)) {
+      throw new Error(`maximumRecordAge must be between 60 and 604800 seconds inclusive, got ${props.maxRecordAge.toSeconds()}`);
+    }
+
+    if (props.retryAttempts && (props.retryAttempts < 0 || props.retryAttempts > 10000)) {
+      throw new Error(`maximumRetryAttempts must be between 0 and 10000 inclusive, got ${props.retryAttempts}`);
+    }
+
+    if ((props.parallelizationFactor || props.parallelizationFactor === 0) && (props.parallelizationFactor < 1 || props.parallelizationFactor > 10)) {
+      throw new Error(`parallelizationFactor must be between 1 and 10 inclusive, got ${props.parallelizationFactor}`);
+    }
+
+    let destinationConfig;
+
+    if (props.onFailure) {
+      destinationConfig = {
+        onFailure: props.onFailure.bind()
+      };
+    }
```

Comment:
Need unit tests for all these.

Reply:
Ah yeah - I forgot to move the tests to the right unit.